### PR TITLE
Enable fetching of go-import headers

### PR DESCRIPTION
**Pull request by n3wscott**
**What this PR does, why we need it**:
Bringing in [buoy](https://github.com/n3wscott/buoy) to knative/test-infra.

This adds go-import parsing by go-import fetch and parse.

### CODE REVIEW

**Comment by chizhg** [comment]
Is the idea here to allow other Knative repos to vendor it, or only `buoy` will vendor it?

If the former, currently only `scripts` in test-infra repo is vendored by other repos, and there was former discussion in https://knative.slack.com/archives/CCSNR4FCH/p1601628568033700 to use a different mechanism to vendor `scripts` instead of using Go mod. So we should make it clear on this.

**Comment by n3wscott** [comment]
@chizhg I am moving Buoy into test-infra. Just in parts so it can be tested and reviewed in parts. 

This is one of many PRs coming in.

File: pkg/golang/goimport.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+func (m *MetaImport) OrgRepo() (string, string) {
+	repoRoot := strings.TrimSuffix(m.RepoRoot, ".git")
+	urlParts := strings.Split(repoRoot, "://")
+	parts := strings.Split(urlParts[len(urlParts)-1], "/")
+	if len(parts) >= 2 {
+		return parts[len(parts)-2], parts[len(parts)-1]
+	}
+	panic(fmt.Errorf("unknown repo root: %s", m.RepoRoot))
```
Comment (mattmoor) [inline pkg/golang/goimport.go:40]: @vagababov will love this line 🤣

Comment (vagababov) [reply pkg/golang/goimport.go:40]: ❌

Comment (n3wscott) [reply pkg/golang/goimport.go:40]: oooh come on.. fine... one sec.

File: pkg/golang/goimport.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+func (m *MetaImport) OrgRepo() (string, string) {
+	repoRoot := strings.TrimSuffix(m.RepoRoot, ".git")
+	urlParts := strings.Split(repoRoot, "://")
+	parts := strings.Split(urlParts[len(urlParts)-1], "/")
+	if len(parts) >= 2 {
+		return parts[len(parts)-2], parts[len(parts)-1]
+	}
+	panic(fmt.Sprint("unknown repo root: ", m.RepoRoot))
```
Comment (vagababov) [inline pkg/golang/goimport.go:40]:
```suggestion
	panic("unknown repo root: " + m.RepoRoot)
```

File: pkg/golang/goimport.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+func metaContent(doc *html.Node, name string) (string, error) {
+	var meta *html.Node
+	var crawler func(*html.Node)
+	crawler = func(node *html.Node) {
```
Comment (vagababov) [inline pkg/golang/goimport.go:46]:
```suggestion
	crawler := func(node *html.Node) {
```

File: pkg/golang/goimport_test.go [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+				t.Errorf("OrgRepo() org = %v, want %v", org, tt.org)
+			}
+			if repo != tt.repo {
+				t.Errorf("OrgRepo() repo = %v, want %v", repo, tt.repo)
+			}
+		})
+	}
+}
+
+func TestMetaImport_OrgRepo_unknownCVS(t *testing.T) {
```
Comment (vagababov) [inline pkg/golang/goimport_test.go:69]:
```suggestion
func TestMetaImport_OrgRepo_UnknownCVS(t *testing.T) {
```

File: pkg/golang/goimport_test.go [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+		}
+	}()
+
+	org, repo := meta.OrgRepo()
+
+	// if we get here, it is a fail.
+	t.Errorf("Expected OrgRepo to panic, got: %s, %s", org, repo)
+}
+
+func Test_metaContent(t *testing.T) {
```
Comment (vagababov) [inline pkg/golang/goimport_test.go:87]:
```suggestion
func TestMetaContent(t *testing.T) {
```

**Review by vagababov — COMMENTED** [review]
🥳

File: pkg/golang/goimport.go (hunk 3 above)
Comment (n3wscott) [reply pkg/golang/goimport.go:46]: can't because :57 deps on a var called `crawler`

Comment (vagababov) [reply pkg/golang/goimport.go:46]: /tableflip

**Review by vagababov — COMMENTED** [review]
/lgtm

**Comment by n3wscott** [comment]
/unhold

@chizhg The intention is not for this to become an import of any other project.

**Pull request merged**